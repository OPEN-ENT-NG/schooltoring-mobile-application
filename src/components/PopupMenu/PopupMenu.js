import React, { Component } from "react";
import PropTypes from "prop-types";
import {
  View,
  UIManager,
  ActionSheetIOS,
  findNodeHandle,
  Platform,
  TouchableOpacity
} from "react-native";
import Icon from "react-native-vector-icons/MaterialIcons";

import I18n from "../../api/I18n";

const ICON_SIZE = 24;

export default class PopupMenu extends Component {
  static propTypes = {
    // array of strings, will be list items of Menu
    actions: PropTypes.arrayOf(PropTypes.string).isRequired,
    onPress: PropTypes.func.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      icon: null
    };
  }

  onError() {
    console.log("Popup Error");
  }

  onPress = () => {
    if (this.state.icon) {
      if (Platform.OS === "android") {
        UIManager.showPopupMenu(
          findNodeHandle(this.state.icon),
          this.props.actions,
          this.onError,
          (action, index) => {
            if (action === "itemSelected") {
              this.props.onPress(index);
            }
          }
        );
      } else {
        let options = [...this.props.actions, I18n.t("cancel")];
        ActionSheetIOS.showActionSheetWithOptions({ options }, index => {
          if (index !== options.length - 1) {
            this.props.onPress(index);
          }
        });
      }
    }
  };

  render() {
    return (
      <View>
        <TouchableOpacity onPress={this.onPress}>
          <Icon
            name="more-vert"
            size={ICON_SIZE}
            color={"grey"}
            ref={this.onRef}
          />
        </TouchableOpacity>
      </View>
    );
  }

  onRef = icon => {
    if (!this.state.icon) {
      this.setState({ icon });
    }
  };
}
