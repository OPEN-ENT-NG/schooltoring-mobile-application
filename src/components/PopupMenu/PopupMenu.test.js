import React from "react";
import { shallow } from "enzyme";

import PopupMenu from "./PopupMenu";

const actions = ["Se connecter", "Se d√©connecter"];
const onPressMenuItem = jest.fn();

const rendered = shallow(
  <PopupMenu actions={actions} onPress={onPressMenuItem} />
);

describe("Render", () => {
  test("PopupMenu should render without crashing", () => {
    expect(rendered).toMatchSnapshot();
  });

  test("PopupMenu should render a 'more-vert' icon", () => {
    expect(
      rendered
        .find("Icon")
        .first()
        .props().name
    ).toEqual("more-vert");
  });
});
