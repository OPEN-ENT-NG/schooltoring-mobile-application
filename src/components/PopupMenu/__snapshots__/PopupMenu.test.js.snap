// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PopupMenu should render without crashing 1`] = `
<View>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Icon
      allowFontScaling={false}
      color="grey"
      name="more-vert"
      size={24}
    />
  </TouchableOpacity>
</View>
`;
